File: Data Structures & Algorithms/duplicate-integer/submission-2.java

class Solution {
    public boolean hasDuplicate(int[] nums) {
        Set<Integer> numsSet = new HashSet<>();
        for (int x=0; x< nums.length;x++){
            Boolean res = numsSet.add(nums[x]);
            if(res==false){
                return true;
            }
        }
        return false;
    }
}

File: Data Structures & Algorithms/permutations-ii/submission-2.py
class Solution:
    def permuteUnique(self, nums: List[int]) -> List[List[int]]:
        vis = [False]*len(nums)
        res = set()
        cur = []
        def dfs():
            if len(cur)==len(nums):
                res.add(tuple(cur))
                return
            # run every element
            for i in range(len(nums)):
                if not vis[i]:
                    vis[i]=True
                    cur.append(nums[i])
                    dfs()
                    vis[i]=False
                    cur.pop()
                    # dfs()
        dfs()
        return [list(x) for x in res]
